File: LanguAI.Backend/LanguAI.Backend/ViewModels/Card/CardListViewModel.cs
using LanguAI.Backend.Core.Enums;
using LanguAI.Backend.Core.Models;

namespace LanguAI.Backend.ViewModels.Card;

public class CardListViewModel
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public Language LearningLanguage { get; set; }

    public Language NativeLanguage { get; set; }

    public string Name { get; set; }

    public DateTime Created { get; set; }

    public DateTime Modified { get; set; }

    public AccessEnum Access { get; set; }

    public List<CardViewModel> CardViewModelList { get; set; } = new List<CardViewModel>();
}

File: LanguAI.Frontend/src/api/models/card-list-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { AccessEnum } from '../models/access-enum';
import { CardViewModel } from '../models/card-view-model';
import { Language } from '../models/language';
export interface CardListViewModel {
  access?: AccessEnum;
  cardViewModelList?: Array<CardViewModel> | null;
  created?: string;
  id?: number;
  learningLanguage?: Language;
  modified?: string;
  name?: string | null;
  nativeLanguage?: Language;
  userId?: number;
}
